hwmgr-plugins/controller/utils/utils.go

```diff
@@ -9,15 +9,86 @@ package utils
 import (
 	"context"
 	"fmt"
+	"reflect"
 	"time"
 
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/client-go/util/retry"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
 // Resource operations
 
+const (
+	UPDATE = "Update"
+	PATCH  = "Patch"
+)
+
+// CreateOrUpdateK8sCR creates/updates/patches an object.
```

Comment:
Duplicate of https://github.com/openshift-kni/oran-o2ims/blob/main/internal/controllers/utils/utils.go#L81 ?

Reply:
Nice find! I had searched the IMS repo and found [`CreateOrUpdate`](https://github.com/openshift-kni/oran-o2ims/blob/main/internal/service/common/clients/k8s/k8s.go#L131) - but this does not support the `patch` operation. [This](https://github.com/openshift-kni/oran-o2ims/blob/main/internal/controllers/utils/utils.go#L81) however does! The name `CreateK8sCR` is a bit misleading. 

I've updated the code to use that!